bdk_rpc/src/lib.rs

```diff
@@ -0,0 +1,377 @@
+                None,
+                Some(page_size),
+                Some(page * page_size),
+                Some(true),
+            )?;
+            let list_len = list.len();
+            // list transaction returns conflicting transactions too. Only consider which are confirmed or in mempool.
+            let list: Vec<ListTransactionResult> = list
+                .into_iter()
+                .filter(|t| {
+                    t.info.confirmations > 0 || client.get_mempool_entry(&t.info.txid).is_ok()
+                })
+                .collect();
+            transactions.extend(list.into_iter());
+            if list_len < page_size {
+                break;
+            }
+            page += 1;
+        }
+        transactions
+    };
+
+    // Filter all transactions already confirmed and included in sparsechain
+    let known_confirms = sparse_chain
+        .iter_tx()
+        .filter_map(|(txid, tx_at_block)| {
+            if tx_at_block.confirmation_time.is_some() {
+                Some(txid)
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<Txid>>();
+
+    let new_txs = list_transactions
+        .into_iter()
+        .filter(|tx_result| !known_confirms.contains(&tx_result.info.txid))
+        .collect::<Vec<_>>();
+
+    let mut transactions = Vec::new();
+
+    for tx_result in new_txs {
+        let txid = tx_result.info.txid;
+        let tx_hex = client.get_transaction(&txid, Some(true))?.hex;
+
+        let tx: Transaction = deserialize(&tx_hex)?;
+
+        let confirmation_time = if let (Some(height), Some(time)) =
+            (tx_result.info.blockheight, tx_result.info.blocktime)
+        {
+            Some(BlockTime { height, time })
+        } else {
+            None
+        };
+
+        transactions.push(TxAtBlock {
+            tx,
+            confirmation_time,
+        })
+    }
+
+    // Find new tip, base tip, and invalidation tips if any
+    let new_tip = client
+        .get_chain_tips()?
+        .iter()
+        .find(|result| result.status == GetChainTipsResultStatus::Active)
+        .map(|tip| BlockId {
+            height: tip.height as u32,
+            hash: tip.hash,
+        })
+        .expect("Active chain tip is always expected");
+
+    let mut invalidate = None;
+    let mut base_tip = None;
+    let known_tips = sparse_chain.iter_checkpoints(..).rev();
+
+    for tip in known_tips {
+        if let Ok(blockhash) = client.get_block_hash(tip.height as u64) {
+            assert!(blockhash == tip.hash);
+            base_tip = Some(tip);
+            break;
+        } else {
+            // TODO: Incase of multi block reorg, handle vector of invalidation
+            invalidate = Some(tip);
+        }
+    }
+
+    let cp = CheckpointCandidate {
+        transactions,
+        base_tip,
+        invalidate,
+        new_tip,
+    };
+
+    // Sync everything
+    match sparse_chain.apply_checkpoint(cp) {
```

Comment:
`sparse_chain` shouldn't be passed in here. The checkpoint should just be returned.